internal/controller/compute/user/user.go

```diff
@@ -0,0 +1,339 @@
+		}).
+		For(&v1alpha1.User{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.UserGroupVersionKind),
+			managed.WithExternalConnecter(&connectorUser{
+				kube:  mgr.GetClient(),
+				usage: resource.NewProviderConfigUsageTracker(mgr.GetClient(), &apisv1alpha1.ProviderConfigUsage{}),
+				log:   l,
+			}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithInitializers(),
+			managed.WithPollInterval(opts.GetPollInterval()),
+			managed.WithTimeout(opts.GetTimeout()),
+			managed.WithCreationGracePeriod(opts.GetCreationGracePeriod()),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+// A connectorUser is expected to produce an ExternalClient when its Connect method
+// is called.
+type connectorUser struct {
+	kube  client.Client
+	usage resource.Tracker
+	log   logging.Logger
+}
+
+// Connect typically produces an ExternalClient by:
+// 1. Tracking that the managed resource is using a ProviderConfig.
+// 2. Getting the managed resource's ProviderConfig.
+// 3. Getting the credentials specified by the ProviderConfig.
+// 4. Using the credentials to form a client.
+func (c *connectorUser) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	svc, err := clients.ConnectForCRD(ctx, mg, c.kube, c.usage)
+	return &externalUser{
+		service: userapi.NewAPIClient(svc, compute.WaitForRequest),
+		log:     c.log,
+	}, err
+}
+
+// externalUser observes, then either creates, updates, or deletes a
+// user in the cloud api. it ensures it reflects a desired state.
+type externalUser struct {
+	// service is the ionos cloud api client to manage users.
+	// see https://api.ionos.com/docs/cloud/v6/#tag/User-management
+	service userapi.Client
+	log     logging.Logger
+}
+
+func connectionDetails(cr *v1alpha1.User, observed ionosdk.User) managed.ConnectionDetails {
+	var details = make(managed.ConnectionDetails)
+	props := observed.GetProperties()
+	details["email"] = []byte(utils.GetStringValue(props.GetEmail()))
+	if cr.Spec.ForProvider.Password != "" {
+		pw := cr.Spec.ForProvider.Password
+		// passwords are sensitive thus should not be part of the cr
+		// they are stored as a secret in the connection details.
+		cr.Spec.ForProvider.Password = ""
+		details[xpv1.ResourceCredentialsSecretPasswordKey] = []byte(pw)
+	}
+	return details
+}
+
+func setStatus(cr *v1alpha1.User, observed ionosdk.User, groupIDs []string) {
+	props := observed.GetProperties()
+	if !observed.HasProperties() {
+		return
+	}
+
+	cr.Status.AtProvider.UserID = utils.GetStringValue(observed.GetId())
+	cr.Status.AtProvider.Active = utils.GetBoolValue(props.GetActive())
+	cr.Status.AtProvider.S3CanonicalUserID = utils.GetStringValue(props.GetS3CanonicalUserId())
+	cr.Status.AtProvider.SecAuthActive = utils.GetBoolValue(props.GetSecAuthActive())
```

Comment:
I saw you changed the if guards using these util functions. However, this doesn't replicate the same computational path.

Here you're mutating fields to their 0 value if the pointer is nil; whereas the previous implementation would not mutate the field if the pointer is nil. 

Not sure this behavior is what's intended.

Reply:
In the previous code, what the cr.Status.AtProvider.SecAuthActive will be in case of props.GetSecAuthActive()=nil?